README.md

```diff
@@ -1,2 +1,83 @@
 # test-automation
-Tools for making our tests easier to run
+
+Tools for making our tests easier to run. Automates setting up a cluster with
+CloudFormation and installs a script which automates setting up citus and everything
+required for testing citus.
+
+# Starting a cluster
+
+You'll need to have installed the (https://aws.amazon.com/cli/)[AWS CLI]. Once that's
+installed you can run something like:
+
+`cloudformation/create-stack.sh -k [your keypair name] -s MyStack -a eu-central-1a`
+
+This will take some time (around 7 minutes) and emit some instructions for connecting to
+the master node.
+
+The name you pass `-s` must be unique. There are more parameters you can pass such as
+`-n`, which changes the number of worker nodes which are launched. If you run
+`create-stack` with no parameters it will tell you more.
+
+# Connecting to the master
+
+The script should have given you a connection string to use. You can also find the
+hostname of the master node in the cloudformation control panel in the "outputs" section.
+
+You'll connect using a string like: `ssh -A ec2-user@ec2-35-156-235-11.eu-central-1.compute.amazonaws.com`.
+
+The "-A" is NOT optional. It is required in order for the master node to be able to ssh
+into the worker nodes and for the nodes to be able to checkout private repos. In
+addition, your ssh-agent should have the key loaded. That means you should not pass the
+`-i` flag into ssh. Instead, run `ssh-add [your pem file]` before
+running ssh.
+
+This is an unfortunate restriction which will hopefully be lifted in the future.
+
+# Example fab commands
+
+Once you have a cluster you can use many different variations of the "fab" command to
+install Citus. Here are some examples:
+
+- `fab --list` will return a list of the tasks you can run.
+- `fab setup.basic_testing`, will create a vanilla cluster with postgres and citus. Once this has run you can simply run `psql` to connect to it.
+- `fab use.citus:v6.0.1 setup.basic_testing` will do the same, but use the tag `v6.0.1` when installing Citus. You can give it any git ref, it defaults to `master`.
+- `fab use.postgres:9.6.1 setup.basic_testing` lets you choose your postgres version.
+- `fab use.citus:v6.0.1 setup.enterprise` will install postgres and the `v6.0.1` tag of the enterprise repo.
```

Comment:
I think there is a lot more to explain in the `README` which could dramatically decrease people asking questions to you over and over :) 

* We should briefly mention what do  `setup`, `use` and `pg` mean for us. In that way, users could comprehend the framework more easily.
* We should be mentioning `valgrind` setup. There is no single word about it.
* We should be mentioning `shard_rebalancer` once we implement it.
* Should show an example or a more detailed explanation of using a closed source project. I wasn't able to run them with following the README. For instance, `citus-enterprise` and `rebalancer` installed on a single cluster.
* `psql postgres` connects to another database. So, should emphasize to simply use `psql` (because, I did that couple of times and got confused)